src/plots/cartesian/graph_interact.js

```diff
@@ -114,6 +114,21 @@ fx.init = function(gd) {
                 xa._length, ya._length, 'ns', 'ew');
 
             maindrag.onmousemove = function(evt) {
+                // This is on `gd._fullLayout`, *not* fullLayout because the reference
+                // changes by the time this is called again.
+                gd._fullLayout._rehover = function() {
+                    if(gd._fullLayout._hoversubplot === plotinfo.id) {
+                        fx.hover(gd, evt, subplot);
+                    }
+                };
+
+                // Track the hovered subplot. This prevents rehover from accidetally
+                // reapplying a hover label after the mouse has left the plot or if
+                // the mouse has entered another subplot.
+                gd._fullLayout._hoversubplot = plotinfo.id;
+
+                gd._fullLayout._rehover();
+
                 fx.hover(gd, evt, subplot);
                 fullLayout._lasthover = maindrag;
                 fullLayout._hoversubplot = subplot;
```

Comment:
Oops. Yeah, you're right. There's still an extra item or two there. Fixing (and looking at the *full* diff when doing so this time)